src/bsd.rs

```diff
@@ -0,0 +1,126 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::{
+    io::Error,
+    mem::{self, size_of},
+    net::IpAddr,
+    os::fd::{AsRawFd, FromRawFd, OwnedFd},
+    ptr, slice, str,
+};
+
+use libc::{
+    getpid, read, rt_msghdr, sockaddr_dl, sockaddr_in, sockaddr_in6, sockaddr_storage, socket,
+    write, AF_INET, AF_INET6, PF_ROUTE, RTAX_IFP, RTAX_MAX, RTA_DST, RTA_IFP, RTM_GET, RTM_VERSION,
+    SOCK_RAW,
+};
+
+use crate::{default_err, next_item_aligned_by_four};
+
+pub fn interface_and_mtu_impl(remote: IpAddr) -> Result<(String, usize), Error> {
+    // Open route socket.
+    let fd = unsafe { OwnedFd::from_raw_fd(socket(PF_ROUTE, SOCK_RAW, 0)) };
```

Comment:
Do I understand correctly that `socket()` can not fail, i.e. does not need to be checked for `-1`?

Reply:
Ah, you're right - I should do the error handling before calling `OwnedFd::from_raw_fd`.